Broken tests for unused import for cloud/services/{network,ec2} in v2.2.4

/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
I stood up envtest and ran `go test ./pkg/...` on `v2.2.4` and I got the following failures for:
```
# sigs.k8s.io/cluster-api-provider-aws/v2/pkg/cloud/services/ec2 [sigs.k8s.io/cluster-api-provider-aws/v2/pkg/cloud/services/ec2.test]
pkg/cloud/services/ec2/bastion_test.go:20:2: "context" imported and not used
FAIL    sigs.k8s.io/cluster-api-provider-aws/v2/pkg/cloud/services/ec2 [build failed]

# sigs.k8s.io/cluster-api-provider-aws/v2/pkg/cloud/services/network [sigs.k8s.io/cluster-api-provider-aws/v2/pkg/cloud/services/network.test]
pkg/cloud/services/network/natgateways_test.go:20:2: "context" imported and not used
pkg/cloud/services/network/vpc_test.go:20:2: "context" imported and not used
FAIL    sigs.k8s.io/cluster-api-provider-aws/v2/pkg/cloud/services/network [build failed]
```

**What did you expect to happen:**
No errors on tests

**Anything else you would like to add:**
It looks like [this commit](https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4510/commits/43b06c8918b86e08819aec9fec8479aa68659111) on the [Bump CAPI to v1.5.0](https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4510) PR, introduced unused imports of `"context"` in:
```
pkg/cloud/services/network/natgateways_test.go
pkg/cloud/services/network/vpc_test.go
pkg/cloud/services/ec2/bastion_test.go
```

That PR got then merged against main, and then got also backported to `release-2.2`, for which a `v2.2.4` got cut today.
At the moment, this is not a problem on `main` as a new PR got merged there which makes use of those previously unused contexts: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4490
Although this still remains an issue on `release-2.2` and on `v2.2.4`.

**Environment:**
- Cluster-api-provider-aws version:  `v2.2.4`
- Kubernetes version: (use `kubectl version`):  local envtest (v1.24`)
- OS (e.g. from `/etc/os-release`): -

--- a/pkg/cloud/services/ec2/bastion_test.go
+++ b/pkg/cloud/services/ec2/bastion_test.go
@@ -17,7 +17,6 @@ limitations under the License.
 package ec2
 
 import (
-	"context"
 	"fmt"
 	"testing"
 

--- a/pkg/cloud/services/network/natgateways_test.go
+++ b/pkg/cloud/services/network/natgateways_test.go
@@ -17,7 +17,6 @@ limitations under the License.
 package network
 
 import (
-	"context"
 	"testing"
 
 	"github.com/aws/aws-sdk-go/aws"

--- a/pkg/cloud/services/network/vpc_test.go
+++ b/pkg/cloud/services/network/vpc_test.go
@@ -17,7 +17,6 @@ limitations under the License.
 package network
 
 import (
-	"context"
 	"testing"
 
 	"github.com/aws/aws-sdk-go/aws"

--- a/pkg/cloud/services/secretsmanager/secret_test.go
+++ b/pkg/cloud/services/secretsmanager/secret_test.go
@@ -17,7 +17,7 @@ limitations under the License.
 package secretsmanager
 
 import (
-	"math/rand"
+	crand "crypto/rand"
 	"sort"
 	"testing"
 
@@ -43,7 +43,7 @@ func TestServiceCreate(t *testing.T) {
 
 	generateBytes := func(g *WithT, count int64) []byte {
 		token := make([]byte, count)
-		_, err := rand.Read(token)
+		_, err := crand.Read(token)
 		g.Expect(err).NotTo(HaveOccurred())
 		return token
 	}

--- a/pkg/cloud/services/ssm/secret_test.go
+++ b/pkg/cloud/services/ssm/secret_test.go
@@ -17,7 +17,7 @@ limitations under the License.
 package ssm
 
 import (
-	"math/rand"
+	crand "crypto/rand"
 	"sort"
 	"strings"
 	"testing"
@@ -45,7 +45,7 @@ func TestServiceCreate(t *testing.T) {
 
 	generateBytes := func(count int64) []byte {
 		token := make([]byte, count)
-		_, err := rand.Read(token)
+		_, err := crand.Read(token)
 		if err != nil {
 			t.Fatalf("error while creating data: %v", err)
 		}

